packages/components/src/button/test/deprecated-props.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * WordPress dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '..';
+
+// The full module mock is necessary to avoid test failures related to
+// calling `console` methods
+jest.mock( '@wordpress/deprecated', () => {
+	return {
+		__esModule: true,
+		default: jest.fn(),
+	};
+} );
```

Comment:
We can use `expect(console).toHaveWarned(...)` instead of mocking, it should also be more semantic too IMO.

Reply:
TIL about `@wordpress/jest-console` ! Thank you for the advice

Done in https://github.com/WordPress/gutenberg/pull/31713/commits/69ed9efd0c6674c8e7d627dc1971e875081bd4fa